app/controllers/tickets_controller.rb

```diff
@@ -0,0 +1,65 @@
+class TicketsController < ApplicationController
+  before_action :set_unit
+  before_action :set_ticket, only: %i[show update destroy]
+  before_action :increment_ticket_num, only: :create
+
+  # GET /units/:unit_hash/tickets
+  def index
+    tickets = @unit.tickets
+    render json: { count: (@unit.tickets_count || 0), tickets: tickets }, status: :ok
+  end
+
+  # GET /units/:unit_hash/tickets/:ticket_num
+  def show
+    render json: { count: 1, tickets: @ticket }, status: :ok
+  end
+
+  # POST /units/:unit_hash/tickets
+  def create
+    # ticket_num relative to unit added as before_create action\
+    new_params = ticket_params.merge(ticket_num: @unit.ticket_num)
+    @ticket = @unit.tickets.new(new_params)
+
+    if @ticket.save
+      render json: { tickets: @ticket }, status: :created
+    else
+      render json: { errors: @ticket.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  # PUT or PATCH /units/:unit_hash/tickets/:ticket_num
+  def update
+    if @ticket.update(ticket_params)
+      render json: {}, status: :no_content
+    else
+      render json: { errors: @ticket.errors.full_messages }, status: :internal_server_error
+    end
+  end
+
+  # DELETE /units/:unit_hash/tickets/:ticket_num
+  def destroy
+    @ticket.destroy
+    render json: {}, status: :no_content
+  end
+
+  private
+
+  def ticket_params
+    params.require(:ticket).permit(:subject, :status, :opened_by, :unit_id, :description, :closed_by, :ticket_num)
```

Comment:
We also have bug severity (high, medium, low) so that field will need to be added in.